src/mrsimulator/spin_system/isotope.py

```diff
@@ -53,7 +55,113 @@ def get_isotope(cls, v, *, values, **kwargs):
         return format_isotope_string(v)
 
     def json(self, **kwargs) -> dict:
-        return self.symbol
+        return get_isotope_data(self.symbol)
+        # return {
+        #     "symbol": self.symbol,
+        #     "spin": self.spin,
+        #     "natural_abundance": self.natural_abundance,
+        #     "gyromagnetic_ratio": self.gyromagnetic_ratio,
+        #     "quadrupole_moment": self.quadrupole_moment,
+        #     "atomic_number": self.atomic_number,
+        # }
+
+    def dict(self, **kwargs) -> dict:
+        return self.json()
+
+    @classmethod
+    def parse_value_from_str_or_dict(cls, val):
```

Comment:
Suggestions to try: rename this to `get_isotope` function which checks the `val` for str, dict, or Isotope types and return the Isotope object.